To: vim-dev@vim.org
Subject: Patch 7.3.004
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.004
Problem:    Crash when using very long regexp. (Peter Odding)
Solution:   Reset reg_toolong. (Carlo Teubner)
Files:	    src/regexp.c


*** ../vim-7.3.003/src/regexp.c	2010-08-15 21:57:29.000000000 +0200
--- src/regexp.c	2010-09-14 10:51:22.000000000 +0200
***************
*** 3538,3543 ****
--- 3538,3544 ----
  
      regline = line;
      reglnum = 0;
+     reg_toolong = FALSE;
  
      /* Simplest case: Anchored match need be tried only once. */
      if (prog->reganch)
*** ../vim-7.3.003/src/version.c	2010-08-17 20:23:18.000000000 +0200
--- src/version.c	2010-09-14 10:52:09.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     4,
  /**/

-- 
I have a watch cat! Just break in and she'll watch.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
